functions/system.bash

```diff
@@ -161,10 +162,10 @@ locale_setting() {
       done
       if ! cond_redirect locale-gen; then echo "FAILED (locale-gen)"; return 1; fi
     fi
-    if ! cond_redirect dpkg-reconfigure --frontend=noninteractive locales; then echo "FAILED (reconfigure locales)"; return 1; fi
+    # shellcheck disable=SC2154
+    syslocale=$system_default_locale
```

Comment:
what if system_default_locale is unset ?
And will /etc/default/locale be adapted to match system_default_locale (if set) ?

Reply:
> what if system_default_locale is unset ?

Then en_US.UTF-8 will be used as defined in line 171:
```shell
if cond_redirect update-locale LANG="${syslocale:-en_US.UTF-8}" LC_ALL="${syslocale:-en_US.UTF-8}" LC_CTYPE="${syslocale:-en_US.UTF-8}" LANGUAGE="${syslocale:-en_US.UTF-8}"; then echo "OK (reboot required)"; else echo "FAILED"; return 1; fi
```

Use the following code snippet to verify that:
```shell

#!/bin/bash
# system_default_locale=de_DE.UTF-8

syslocale=$system_default_locale

echo ${syslocale:-en_US.UTF-8}
```

> And will /etc/default/locale be adapted to match system_default_locale (if set) ?

`update-locale` automatically updates `/etc/default/locale` on every call.

Reply:
but it's only called before your new line syslocale=$system_default_locale so does not have the value

Use defaults on assignments i.e. X=${Y:-Z}, can even be X=${Y:-${Z:-"z"}}

Reply:
> but update-locale is only called before your new line syslocale=$system_default_locale so does not have the value by then

No, see:
```shell
…

    # shellcheck disable=SC2154
    syslocale=$system_default_locale
  fi

  if cond_redirect update-locale LANG="${syslocale:-en_US.UTF-8}" LC_ALL="${syslocale:-en_US.UTF-8}" LC_CTYPE="${syslocale:-en_US.UTF-8}" LANGUAGE="${syslocale:-en_US.UTF-8}"; then echo "OK (reboot required)"; else echo "FAILED"; return 1; fi
…
```

Reply:
> Use defaults on assignments i.e. X=${Y:-z}, can even be X=${Y:-${Z:-z}}

In the line `syslocale=$system_default_locale`?

Reply:
yes and everywhere where appropriate

Reply:
In the update-locale statement, for the `syslocale` var it is not required. (If interactive, an error is thrown if syslocale cant be assigned, and if unattened, syslocale gets a default from the assignment to system_default_locale.

Can I remove it?

Reply:
> if unattened, syslocale gets a default from the assignment to system_default_locale.

which is not guaranteed to be set as it's user-defined.

At some place you have to provide a default to apply to all use cases.

Reply:
See my latest commit, I assign a default if system_default_locale is not set and we are in unattended mode.
If we are in interactive mode, syslocal can not be empty, because the function exits with an error if the assignment fails.